### ISSUE TITLE
[BUG] Cloud cover in AIFS not properly computed

### ISSUE THREAD

**Issue by guidocioni** [open-meteo/open-meteo#702]
At the beginning I was going to bash the AI revolution by saying that it doesn't really predict cloud cover well :D, then I realized cloud cover is diagnosed based on relative humidity, so I think there may be something fishy going on when computing it. 

For example, this is the cloud cover diagnosed for the main models (notice the large deviation of `AIFS`)

![7e13e09c-aec6-4c33-8e8e-086d09cd84a7](https://github.com/open-meteo/open-meteo/assets/12760310/7d3d5ea4-f4d6-4c45-90e8-7f2499032175)

and these are the relative humidity values for the same models

![757552ab-232c-4064-8418-1ccc8e87fd8d](https://github.com/open-meteo/open-meteo/assets/12760310/dfe3477c-ef26-4671-85e6-3803071ecfee)

I think there must be something wrong when computing cloud cover for `AIFS`, because the RH spread is not as large as the one for cloud cover.

**Comment by guidocioni** [comment]
> AIFS and IFS cloud cover values are both calculated from pressure level relative humidity. AIFS uses specific humidity, but this conversion step is trivial. Afterwards it is approximating cloud cover from relative humidity on specific pressure levels:
> 
> https://github.com/open-meteo/open-meteo/blob/eadf59ab484e8a1c8b9be87292b3f130caba71e3/Sources/App/Helper/Meteorology.swift#L325-L335
> 
> On the first sight, it does not completely wrong. Just to better understand your graphs: Which relative humidty and cloud cover is shown here? The 2 m relative humidity has no effect on cloud cover. Only pressure level cloud cover is used.

Hey Patrick,
first of all you're totally right about RH. I plotted the one at 2m, no idea why. Let's forget about that ;)
Still, the cloud cover value in the first plot is straight from the API so I haven't done anything with that. You can check directly in the docs https://open-meteo.com/en/docs#hourly=cloud_cover&models=ecmwf_ifs025,ecmwf_aifs025,gfs_global,icon_global and you'll see that the values of AIFS are quite far from all the other models.

Is the total cloud cover coming directly from the model as diagnostic variable or do you obtain it as vertical integration based on the cloud cover in pressure coordinates that you diagnose yourself? 

I was just checking the vertical distribution of cloud cover for different models. 
This is for the "normal" IFS 
![newplot](https://github.com/open-meteo/open-meteo/assets/12760310/ed2d8326-3971-4a8d-a6b0-4f334d1feae4)
And this is for AIFS
![newplot (2)](https://github.com/open-meteo/open-meteo/assets/12760310/0f672525-be46-4e53-a414-5fec46bda6ef)

I think the difference between the two models is ok, because this is what comes from the comparison between ICON and GFS for the same period. 
![newplot (5)](https://github.com/open-meteo/open-meteo/assets/12760310/0ee88987-8bef-4bb6-a5ce-fbf5631e8d47)
![newplot (4)](https://github.com/open-meteo/open-meteo/assets/12760310/222bc754-caad-409b-8dee-309bb3b7be66)

Do you think the differences we see are only from the model itself? As a matter of fact total cloud cover in ICON as GFS is also quite different... Maybe it's not a bug after all :D

**Comment by patrick-zippenfenig** [comment]
Thanks for the contour plots. They are very useful. IFS and AIFS use the same conversion from pressure level humidity to fractional cloud cover (Sundqvist et al.). 

This approximation does not account for condensation nuclei and other effects to form clouds. This results in barely any clouds in the higher atmosphere. Some weather models like GFS offer fractional cloud cover on pressure level directly. Although GFS tends to forecast too many clouds on lower levels (IMHO).

For low cloud cover in AIFS:
1. It could just be forecast too low
2. The [specific to relative humidity calculation](https://github.com/open-meteo/open-meteo/blob/eadf59ab484e8a1c8b9be87292b3f130caba71e3/Sources/App/Helper/Meteorology.swift#L180-L196) returns too low results
3. The interpolation from 6-hourly data to 1-hourly is unfavourable for cloud cover calculation

Did you observe the same low cloud cover forecast from AIFS for other locations as well? It is "always" too low?

**Comment by guidocioni** [comment]
> Thanks for the contour plots. They are very useful. IFS and AIFS use the same conversion from pressure level humidity to fractional cloud cover (Sundqvist et al.).
> 
> This approximation does not account for condensation nuclei and other effects to form clouds. This results in barely any clouds in the higher atmosphere. Some weather models like GFS offer fractional cloud cover on pressure level directly. Although GFS tends to forecast too many clouds on lower levels (IMHO).
> 
> For low cloud cover in AIFS:
> 
> 1. It could just be forecast too low
> 2. The [specific to relative humidity calculation](https://github.com/open-meteo/open-meteo/blob/eadf59ab484e8a1c8b9be87292b3f130caba71e3/Sources/App/Helper/Meteorology.swift#L180-L196) returns too low results
> 3. The interpolation from 6-hourly data to 1-hourly is unfavourable for cloud cover calculation
> 
> Did you observe the same low cloud cover forecast from AIFS for other locations as well? It is "always" too low?

Yes I did observe the same behavior for other locations. It seems that AIFS has always less clouds, although I haven't really had the time to check if this due to the lack of low, middle of high clouds.

For ICON you get cloud cover values directly as well as for GFS right?

I think the difference in cloud cover between IFS and AIFS is an interesting topic to investigate, but given that you diagnose these based on relative humidity (as they're not available in the output) it may be hard to draw some conclusions..

P.S. You can experiment yourself for different locations in my app. I produced these plots using the "vertical" page and just turning off  the temperature, geopotential and winds traces in the legend.

**Comment by patrick-zippenfenig** [comment]
> For ICON you get cloud cover values directly as well as for GFS right?

Unfortunately ICON only offers cloud cover fraction for model levels, but not for pressure levels. Therefore I use the same conversion from humidity to fractional cloud cover (Sundqvist et al.).

**Comment by guidocioni** [comment]
> > For ICON you get cloud cover values directly as well as for GFS right?
> 
> Unfortunately ICON only offers cloud cover fraction for model levels, but not for pressure levels. Therefore I use the same conversion from humidity to fractional cloud cover (Sundqvist et al.).

ICON-EU does have cloud cover on pressure levels, just FYI. Yes, unfortunately the other models do not offer that but another possibility would be to do model level to pressure levels interpolation instead. But of course I understand this is much more data to process...

Regarding the difference with AIFS and other models do we then agree that is due only to the model? Then we can close the issue

**Comment by guidocioni** [comment]
Seems that AI models have completely different cloud covers :D 

**IFS vs AIFS**
![ifs](https://github.com/open-meteo/open-meteo/assets/12760310/085d6595-5814-476f-ba2f-022893d5627e)

**GFS vs GRAPHCAST**
![gfs](https://github.com/open-meteo/open-meteo/assets/12760310/9eccb564-6263-4cf5-b832-8e9697869a9f)

**Comment by patrick-zippenfenig** [comment]
Hm... GraphCast and AIFS are fairly similar, but I am now getting more and more worried, that I should change the cloud parameterisation....
